src/Microsoft.CorrelationVector/CorrelationVectorV2.cs

```diff
@@ -0,0 +1,404 @@
+            {
+                CorrelationVectorV2.Validate(correlationVector, version);
+            }
+
+            byte[] entropy = new byte[parameters.EntropyBytes];
+            rng.NextBytes(entropy);
+
+            ulong value = (ulong)(DateTime.UtcNow.Ticks >> parameters.TicksBitsToDrop);
+            for (int i = 0; i < parameters.EntropyBytes; i++)
+            {
+                value = (value << 8) | Convert.ToUInt64(entropy[i]);
+            }
+
+            // Generate a bitmask and mask the lower TotalBits in the value.
+            // The mask is generated by (1 << TotalBits) - 1. We need to handle the edge case
+            // when shifting 64 bits, as it wraps around.
+            value &= (parameters.TotalBits == 64 ? 0 : (ulong)1 << parameters.TotalBits) - 1;
+
+            string s = unchecked((uint)value).ToString();
+            if (parameters.TotalBits > 32)
+            {
+                s = string.Concat((value >> 32).ToString(), ".", s);
+            }
+
+            string baseVector = string.Concat(correlationVector, ".", s);
+            if (CorrelationVectorV2.IsOversized(baseVector, 0, version))
+            {
+                return CorrelationVectorV2.Parse(correlationVector + CorrelationVectorV2.TerminationSign);
+            }
+
+            return new CorrelationVectorV2(baseVector, 0, version, false);
+        }
+
+        /// <summary>
+        /// Creates a new correlation vector by parsing its string representation
+        /// </summary>
+        /// <param name="correlationVector">correlationVector</param>
+        /// <returns>CorrelationVector</returns>
+        public new static CorrelationVectorV2 Parse(string correlationVector)
+        {
+            if (!string.IsNullOrEmpty(correlationVector))
+            {
+                int p = correlationVector.LastIndexOf('.');
+                bool immutable = CorrelationVectorV2.IsImmutable(correlationVector);
+                if (p > 0)
+                {
+                    string extensionValue = immutable ? correlationVector.Substring(p + 1, correlationVector.Length - p - 1 - CorrelationVectorV2.TerminationSign.Length)
+                        : correlationVector.Substring(p + 1);
+                    int extension;
+                    if (int.TryParse(extensionValue, out extension) && extension >= 0)
+                    {
+                        return new CorrelationVectorV2(correlationVector.Substring(0, p), extension, CorrelationVectorV2.InferVersion(correlationVector, false), immutable);
+                    }
+                }
+            }
+
+            return new CorrelationVectorV2();
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="CorrelationVectorV2"/> class. This
+        /// should only be called when no correlation vector was found in the message
+        /// header.
+        /// </summary>
+        public CorrelationVectorV2()
+            : this(CorrelationVectorVersion.V1)
```

Comment:
This is incorrect. If we are creating a new instance of V2, we should only use the V2 version